src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3042,11 +3042,13 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
                                ? YMM_REGSIZE_BYTES
                                : XMM_REGSIZE_BYTES;
 
+        bool zeroing = false;
         if (src->gtSkipReloadOrCopy()->IsIntegralConst(0))
         {
             // If the source is constant 0 then always use xorps, it's faster
             // than copying the constant from a GPR to a XMM register.
             emit->emitIns_R_R(INS_xorps, EA_ATTR(regSize), srcXmmReg, srcXmmReg);
+            zeroing = true;
         }
         else
         {
```

Comment:
Just noting since I happened to look here... We're initializing from a `srcIntReg` and we probably would have been better off if we did the "can use simd copy" check in lowering instead and opted to lower to a `gtNewSimdCreateBroadcastNode`

This will do the most efficient thing of creating a `GenTreeVecCon` (which includes efficiently getting a `zero` or `allbitsset` node rather than pulling from memory) -or- it will do the "most efficient" broadcast of the value into the `xmm` register (`movd` followed by `vpbroadcastd` or `pshufd`, typically).

What we're doing right now "works", but it's quite a bit more expensive than it needs to be and won't see improvements from other SIMD opts we enable over time.

Reply:
I'd leave this up for grabs since it requires a broader refactoring to do the whole thing in Lower. This PR only slightly improves codegen where it's too late to work with SIMD gen trees